### ISSUE TITLE
Refactoring and UI Rewrite

### ISSUE THREAD

**Issue by vknabel** [vknabel/puffery#65]
Puffery has initially been developed for iOS 13 and was ported to iOS 14, while I was still learning SwiftUI. 

Though iOS 15 provides whole new opportunities and far less workarounds for certain features.

Especially the macOS version would've been tricky to make.

With this rewrite, I try to accomplish more platform variety, getting rid of old approaches and personally experienced dead-ends in terms of coding experiments.

The new version will use the same architecture for all parts. It shall be possible to display more than just a limited amount of messages (originally due to performance issues in iOS 13). To correctly power this, we want to rely on Core Data and standard tooling as much as possible.

In summary: this update will hopefully be released with iOS 15 and will immediately drop support for all previous versions.
